src/lib.rs

```diff
@@ -25,8 +26,16 @@ pub use baseboard::BaseBoard;
 pub mod processor;
 pub use processor::Processor;
 
+///
+/// An SMBIOS `Entry` structure.
+///
+/// The SMBIOS `Entry` structure is used to verify that a set of SMBIOS tables exist
+/// in memory and what version of the SMBIOS specification should be used to
+/// access the tables.
+///
 #[repr(C)]
 #[repr(packed)]
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
 pub struct Entry {
```

Comment:
Is 'Entry' and 'Structure' here a standard name?

Reply:
`EntryPoint` might be a better name then. I've been reading it as in an entry in a list. That would line up well with `EntryPoint::search` as well.

Reply:
Makes sense. I can make that change.